services/director-v2/src/simcore_service_director_v2/models/dynamic_services_scheduler.py

```diff
@@ -475,6 +476,14 @@ def get_proxy_endpoint(self) -> AnyHttpUrl:
         ),
     ] = None
 
+    product_api_base_url: Annotated[
+        str | None,
+        BeforeValidator(lambda v: f"{AnyHttpUrl(v)}"),
+        Field(
+            description="Base URL for the current product's API.",
+        ),
+    ] = None
```

Comment:
should it really be nullable?

Reply:
@GitHK there are some mocks files (i.e. `fake_scheduler_data_compose_spec.json`, `fake_scheduler_data.json` and  `legacy_scheduler_data_format.json`) that would require a change as well... and I don't know where they come from

Reply:
The ideas if the tests is to avoid breaking the `legacy_scheduler_data_format.json`.
Usually we release with all dynamic services shut down so it should not be an issue even if this format changes.
I would keep it as is